htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -272,6 +272,15 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 				$i = 0;
 				if ($num > 0)
 				{
+					if ( $product->status_batch == 2 && ($type == 0 || $type == 3) ) {
```

Comment:
The table product_lot is a memory table to store properties of a lot. Not to store we have this lot in stock or not.
Having a record into this table should not block any stock movement on this lot.
We may have a record into this table (as a memory because lot did exists in the past) and have a quantity in stock of 0. So trying to create a stock movement of +1 is legitimate and should not be blocked.
Only test on final quantity to be sure we have 0 or 1 at end of transaction should block.

Reply:
i agree

Reply:
but where do you check the serail number is not already used for the product ?